compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/BlendedManifest.java

```diff
@@ -132,11 +133,12 @@ public static BlendedManifest from(DependencyManifest dependencyManifest,
                     }
                 }
             } else {
+                Collection<String> moduleNames = existingDepOptional.isPresent()?
+                        existingDepOptional.get().modules : Collections.emptyList();
                 depContainer.add(depInPkgManifest.org(), depInPkgManifest.name(), new Dependency(
                         depInPkgManifest.org(), depInPkgManifest.name(), depInPkgManifest.version(),
                         DependencyRelation.UNKNOWN, REPOSITORY_NOT_SPECIFIED,
-                        moduleNames(new DependencyManifest.Package(depInPkgManifest.name(), depInPkgManifest.org(),
-                                depInPkgManifest.version())), DependencyOrigin.USER_SPECIFIED));
```

Comment:
This `moduleNames(new DependencyManifest.Package(depInPkgManifest.name(), depInPkgManifest.org(), depInPkgManifest.version()))` always returns an empty list of module names.